rules/prefer-error-cause.js

```diff
@@ -0,0 +1,388 @@
+/* eslint-disable capitalized-comments */
+/* eslint-disable brace-style */
+'use strict';
+const esquery = require('esquery');
+const {isCommaToken} = require('eslint-utils');
+const _ = require('lodash');
+const {matches, methodCallSelector} = require('./selectors/index.js');
+const {appendArgument, replaceReferenceIdentifier} = require('./fix/index.js');
+const getReferences = require('./utils/get-references.js');
+
+const ERROR = 'error';
+const SUGGESTION = 'suggestion';
+
+const messages = {
+	[ERROR]: 'Prefer specifying an old error as `cause` option where rethrowing the error.',
+	[SUGGESTION]: 'Specify `cause` option to Error\'s constructor.',
+};
+
+const catchClauseSelector = matches([
+	[
+		'CatchClause',
+		':has(ThrowStatement)',
+	].join(''),
+]);
+
+const promiseCatchSelector = matches([
+	[
+		methodCallSelector({method: 'catch', argumentsLength: 1}),
+		':has(ThrowStatement)',
+	].join(''),
+]);
+
+const peekTop = array => {
+	if (_.isEmpty(array)) {
+		return undefined;
+	}
+
+	return array[array.length - 1];
+};
```

Comment:
This function is meaningless. Use `array[array.length - 1]` directly. Or `const peekTop = array => array[array.length - 1];`